core/src/main/java/com/valtech/aem/saas/core/http/request/SearchRequestDelete.java

```diff
@@ -0,0 +1,41 @@
+package com.valtech.aem.saas.core.http.request;
+
+import java.util.Collections;
+import java.util.List;
+import javax.servlet.http.HttpServletResponse;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.HttpUriRequest;
+
+/**
+ * Represents a DELETE search request with request body. It requires the request uri and the payload in form of {@link
+ * HttpEntity}.
+ */
+@Slf4j
+@Builder
+public class SearchRequestDelete implements SearchRequest {
+
+  @NonNull
+  private final String uri;
+  private final HttpEntity httpEntity;
+
+  @Override
+  public HttpUriRequest getRequest() {
+    if (StringUtils.isBlank(uri)) {
```

Comment:
do we need this check if the uri is @NotNull?

Reply:
Yes, uri can be an emprty string as well, in which case it will not pass this check